src/data_types/w3c/credential.rs

```diff
@@ -73,356 +55,120 @@ pub struct CredentialSubject {
-pub enum CredentialStatusType {
-    AnonCredsCredentialStatusList2023,
-    Other(String),
-}
-
-impl Display for CredentialStatusType {
-    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        match self {
-            CredentialStatusType::AnonCredsCredentialStatusList2023 => {
-                f.write_str("AnonCredsCredentialStatusList2023")
-            }
-            CredentialStatusType::Other(other) => f.write_str(other),
-        }
-    }
-}
-
-impl From<&str> for CredentialStatusType {
-    fn from(value: &str) -> Self {
-        match value {
-            "AnonCredsCredentialStatusList2023" => {
-                CredentialStatusType::AnonCredsCredentialStatusList2023
-            }
-            other => CredentialStatusType::Other(other.to_string()),
-        }
-    }
-}
-
-impl Serialize for CredentialStatusType {
-    fn serialize<S>(&self, serializer: S) -> core::result::Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-    {
-        Value::String(self.to_string()).serialize(serializer)
-    }
-}
-
-impl<'de> Deserialize<'de> for CredentialStatusType {
-    fn deserialize<D>(deserializer: D) -> core::result::Result<Self, D::Error>
-    where
-        D: Deserializer<'de>,
-    {
-        Value::deserialize(deserializer)
-            .map_err(de::Error::custom)?
-            .as_str()
-            .map(CredentialStatusType::from)
-            .ok_or_else(|| de::Error::custom("Cannot parse credential status type"))
-    }
+pub enum CredentialProof {
+    DataIntegrityProof(DataIntegrityProof),
+    NonAnonCredsDataIntegrityProof(NonAnonCredsDataIntegrityProof),
 }
 
-impl Default for CredentialStatusType {
-    fn default() -> Self {
-        CredentialStatusType::AnonCredsCredentialStatusList2023
-    }
-}
-
-impl CredentialStatus {
-    pub fn new(id: RevocationRegistryDefinitionId) -> CredentialStatus {
-        CredentialStatus {
-            type_: CredentialStatusType::AnonCredsCredentialStatusList2023,
-            id,
-        }
-    }
-}
-
-#[derive(Debug, Clone, Default, PartialEq, Eq, Serialize, Deserialize)]
-#[serde(rename_all = "camelCase")]
-pub struct CredentialSchema {
-    #[serde(rename = "type")]
-    pub type_: CredentialSchemaType,
-    pub definition: CredentialDefinitionId,
-    pub schema: SchemaId,
-    #[serde(default)]
-    pub encoding: CredentialValuesEncoding,
-}
-
-impl CredentialSchema {
+impl W3CCredential {
     pub fn new(
-        schema: SchemaId,
-        definition: CredentialDefinitionId,
-        encoding: CredentialValuesEncoding,
-    ) -> CredentialSchema {
-        CredentialSchema {
-            type_: CredentialSchemaType::AnonCredsDefinition,
-            definition,
-            schema,
-            encoding,
+        issuer: IssuerId,
+        attributes: CredentialAttributes,
+        proof: DataIntegrityProof,
+        version: Option<&VerifiableCredentialSpecVersion>,
+    ) -> Self {
+        let time = Utc::now();
```

Comment:
validFrom is not required in v2, so I think we should leave it out in this case. Adding it for the v1 one version is just a 'hack' I think to make it compliant with the data model